src/dataframe/dataframe.jl

```diff
@@ -1330,17 +1433,37 @@ julia> repeat(df, 2)
 """
 function repeat!(df::DataFrame, count::Integer)
     count < 0 && throw(ArgumentError("count must be non-negative"))
-    return mapcols!(x -> repeat(x, Int(count)), df)
+    cols = _columns(df)
+    for (i, col) in enumerate(cols)
+        col_new = repeat(col, count)
+        firstindex(col_new) != 1 && _onebased_check_error(i, col_new)
+        cols[i] = col_new
+    end
+    _drop_all_nonnote_metadata!(df)
+    return df
 end
 
 # This is not exactly copy! as in general we allow axes to be different
+# Also no table metadata needs to be copied as we use _replace_columns!
+# only in situations when table metadata is kept
```

Comment:
The sentence means:
if we call `_replace_columns` on `df` then we do not need to touch table-level metadata as we know that appropriate table-level metadata is already in `df` (because `newdf` is derived from `df` anyway, so table metadata does not need to be handled).

It is not copied by the caller. It is in `df` from the very beginning. This function is used in `select!` and `transform!`. These functions update source data frame. And this source data frame already has its table-level metadata so we do not need to touch it.

How would you phrase it?

Reply:
done